packages/amplify_datastore_plugin_interface/lib/src/types/temporal/temporal_datetime.dart

```diff
@@ -0,0 +1,112 @@
+import 'temporal.dart';
+
+class TemporalDateTime {
+  DateTime _dateTime;
+  int _nanoseconds;
+  Duration _offset;
+
+  ///
+  static TemporalDateTime now() {
+    return TemporalDateTime(DateTime.now());
+  }
+
+  ///
+  TemporalDateTime(DateTime dateTime) {
+    dateTime = dateTime.toUtc();
+    _dateTime = DateTime.utc(dateTime.year, dateTime.month, dateTime.day,
+        dateTime.hour, dateTime.minute, dateTime.second);
+    _nanoseconds =
+        (dateTime.millisecond * 1000000) + (dateTime.microsecond * 1000);
+  }
+
+  ///
+  TemporalDateTime.withOffset(DateTime dateTime, Duration offset) {
+    dateTime = dateTime.toUtc();
+    _dateTime = DateTime.utc(dateTime.year, dateTime.month, dateTime.day,
+        dateTime.hour, dateTime.minute, dateTime.second);
+    _nanoseconds =
+        (dateTime.millisecond * 1000000) + (dateTime.microsecond * 1000);
+
+    if (offset.inDays > 0) {
+      throw new Exception("Cannot have an offset in days (hh:mm:ss)");
+    }
+
+    _offset = offset;
+  }
+
+  // DateTime Z is required
```

Comment:
For AWSDateTime specs it says: 
`The time zone offset is compulsory for this scalar. The time zone offset must either be Z (representing the UTC time zone) or be in the format ±hh:mm:ss. 
`
https://docs.aws.amazon.com/appsync/latest/devguide/scalars.html

Reply:
I feel like there's something to be said about hardcoding AppSync type specific requirements at the Temporal.DateTime level, like is this AppSync.AWSDateTime or Temporal.DateTime? I'm not sure what the solution is since even if we relax the validation because iso8601 spec allows it, it would then fail at the API call. perhaps it's up to the API plugin to validate the iso8601String of Temporal.DateTime's. This would still be troublesome since developers are actually calling DataStore and so SQL wouldn't reject it, then only in the sync engine would it be rejected.

Reply:
Good point, though isn't the entire purpose of Temporal.DateTime to be stored to AppSync?

Reply:
Could this type also be relevant for the API category, not just Datastore? If so we could move it to amplify_core?